olp-cpp-sdk-authentication/include/olp/authentication/AuthorizeResult.h

```diff
@@ -42,9 +42,17 @@ enum class DecisionType { kAllow, kDeny };
 class AUTHENTICATION_API ActionResult {
  public:
   /**
-   * @brief Represents the permission pair with the action and policy decision.
+   * @brief Represents the permission with the action, policy decision and
+   * associated resource.
    */
-  using Permissions = std::pair<std::string, DecisionType>;
+  struct Permission {
```

Comment:
This breaks a little our normal flow. The normal flow is that these structs look like swagger generated.
This includes getters/setter for all members and a separate file.